src/input/type_record.cc

```diff
@@ -62,6 +63,23 @@ TypeBase::TypeHash Default::content_hash() const
 }
 
 
+bool Default::check_validity(const TypeBase &type) const
+{
+	if ( !has_value_at_declaration() ) return false;
+
+	try {
+		istringstream is(value_);
+		Input::ReaderToStorage reader;
+		reader.read_stream(is, type, FileFormat::format_JSON);
+		return true;
+	} catch ( Input::ReaderToStorage::ExcNotJSONFormat &e ) {
+		THROW( ExcWrongDefault() << EI_DefaultStr( value_ ) << EI_TypeName(type.type_name()));
+	} catch ( Input::ReaderToStorage::ExcInputError &e ) {
+		THROW( ExcWrongDefault() << EI_DefaultStr( value_ ) << EI_TypeName(type.type_name()));
+	}
+}
+
+
 
```

Comment:
Add a pointer to storage into Default class.
Add method get_storage() (automatically calls check_validity if the storage is NULL)
Use the storage in ReaderToStorage and in Abstract::get_default_descendant.